web/src/compose.js

```diff
@@ -534,6 +534,31 @@ export function initialize() {
         },
     );
 
+    $("body").on(
+        "click",
+        `.${CSS.escape(
+            compose_banner.CLASSNAMES.compose_in_search_view,
+        )} .main-view-banner-action-button`,
+        (event) => {
+            event.preventDefault();
+            const stream_name = compose_state.stream_name();
+            const recipient = compose_state.private_message_recipient();
+            if (stream_name) {
+                narrow.activate([
+                    {operator: "stream", operand: stream_name},
+                    {operator: "topic", operand: compose_state.topic()},
+                ]);
+            } else {
+                narrow.activate([{operator: "dm", operand: recipient}]);
+            }
+            $(`.${CSS.escape(compose_banner.CLASSNAMES.compose_in_search_view)}`).remove();
+        },
+    );
+
+    $("body").on("mouseup", ".topic-name", () => {
+        $(`.${CSS.escape(compose_banner.CLASSNAMES.compose_in_search_view)}`).remove();
+    });
```

Comment:
What does his handler do? This `topic-name` selector is way too vague of a selector -- one could imagine all sorts of unrelated elements using that class name.

Reply:
Oh, I see, you're trying to do something when the view changes -- yeah this is reliant on that particular click handler, but there's lots of other ways to change a view that should hide this banner.

I think you want to just clear this banner from `narrow.handle_middle_pane_transition` -- but to do that nicely, you'll want to define a `compose_banner.hide_compose_in_search_banner()` function to avoid internal compose component details/selectors in the `narrow.js` code.